File: Runtime Method Swizzling/Runtime Method Swizzling/ViewController.m
//
//  ViewController.m
//  Runtime Method Swizzling
//
//  Created by BoBo on 17/2/17.
//  Copyright © 2017年 Li Sen. All rights reserved.
//

#import "ViewController.h"
#import "SecondViewController.h"

@interface ViewController ()

@end

@implementation ViewController

//通过Method Swizzling来把系统的方法交换成我们自己的方法,从而给系统方法添加一些我们想要的功能
- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (void)viewWillAppear:(BOOL)animated {
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)pushButtonDidClicked:(id)sender {
    SecondViewController *secondVC = [[SecondViewController alloc] init];
    [self presentViewController:secondVC animated:true completion:nil];
}

@end

File: WebView与JS交互/WebView与JS交互/ViewController.swift
//
//  ViewController.swift
//  WebView与JS交互
//
//  Created by BoBo on 17/3/15.
//  Copyright © 2017年 Li Sen. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var triggerBtn: UIButton!
    @IBOutlet weak var webView: UIWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        //从本地加载html
        let path: String = Bundle.main.path(forResource: "index", ofType: "html")!
        webView.loadRequest(URLRequest(url: URL(fileURLWithPath: path)))
        
        //MARK: - app调用js方法
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    //直接调用JS的方法
    @IBAction func eventButtonDidClick(_ sender: UIButton) {
       webView.stringByEvaluatingJavaScript(from: "hi()")
    }
}

